dashboard/app/controllers/registrations_controller.rb

```diff
@@ -422,6 +426,11 @@ def email_preference_params(source, form_kind)
       )
   end
 
+  def email_conflict_params
+    params.require(:email)
+    params.require(:provider)
```

Comment:
Does this work? I'm reading this as checking the presence of the `:email` and `:provider` params, but only returning the value of the `:provider` param.

Reply:
Yep! `params.permit` returns something that we use and care about, but `params.require` just ensures that the specific parameter or parameters are present.

That being said, this could more clearly be written as `params.require([:email, :provider])`

Reply:
So, do we not care about the return value of this function at all? Because it looks like we're using it, above.

Reply:
oh, good catch; we don't care, so I'll stop using it.